### PR TITLE
Added pandasgui package

### PR DESCRIPTION
**Pull request by adamerose**
Checklist
- [✓] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [✓] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [✓] Source is from official source
- [✓] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [✓] If static libraries are linked in, the license of the static library is packaged.
- [✓] Build number is 0
- [✓] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [✓] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by danschef** [comment]
@ocefpaf @mingwandroid I ping you here as we seem to have a similar problem to this one [here](https://github.com/conda-forge/pygridgen-feedstock/issues/10). The build process fails for Linux with `ImportError: libGL.so.1: cannot open shared object file: No such file or directory`. I already tried to add libgl1-mesa-glx to the host requirements in case of Linux but it does not seem to work.

Do you have any suggestion what is wrong with this recipe? The OSX build also fails with another error but I guess this does not matter in case of a noarch package.

**Comment by ocefpaf** [comment]
@danschef please take a look at https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl

**Comment by danschef** [comment]
Thanks , I updated the recipe accordingly (looks like we can`t build it as a noarch package anymore). However, now there another library missing in the Linux build:

```
import: 'pandasgui'
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/pandasgui_1605539216465/test_tmp/run_test.py", line 2, in <module>
    import pandasgui
  File "/home/conda/staged-recipes/build_artifacts/pandasgui_1605539216465/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/site-packages/pandasgui/__init__.py", line 1, in <module>
    from pandasgui.gui import show
  File "/home/conda/staged-recipes/build_artifacts/pandasgui_1605539216465/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/site-packages/pandasgui/gui.py", line 14, in <module>
    from pandasgui.widgets.dataframe_explorer import DataFrameExplorer
  File "/home/conda/staged-recipes/build_artifacts/pandasgui_1605539216465/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/site-packages/pandasgui/widgets/dataframe_explorer.py", line 9, in <module>
    from pandasgui.widgets.grapher import Grapher
  File "/home/conda/staged-recipes/build_artifacts/pandasgui_1605539216465/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/site-packages/pandasgui/widgets/grapher.py", line 16, in <module>
    from pandasgui.widgets.plotly_viewer import PlotlyViewer
  File "/home/conda/staged-recipes/build_artifacts/pandasgui_1605539216465/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/site-packages/pandasgui/widgets/plotly_viewer.py", line 32, in <module>
    raise e
  File "/home/conda/staged-recipes/build_artifacts/pandasgui_1605539216465/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/site-packages/pandasgui/widgets/plotly_viewer.py", line 20, in <module>
    from PyQt5 import QtWebEngineWidgets
ImportError: libXcomposite.so.1: cannot open shared object file: No such file or directory
```

File: recipes/pandasgui/meta.yaml [hunk 1]
```diff
@@ -12,14 +12,22 @@ source:
...
   script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
+  build:
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - {{ cdt('mesa-dri-drivers') }}  # [linux]
+    - {{ cdt('libselinux') }}  # [linux]
+    - {{ cdt('libxdamage') }}  # [linux]
+    - {{ cdt('libxxf86vm') }}  # [linux]
+    - {{ cdt('libxext') }}     # [linux]
```
Comment (ocefpaf) [inline recipes/pandasgui/meta.yaml:25]: You can still be noarch and you probably don't need all of them. Please try adding only the yum requirements with `mesa-libGL` first.

Comment (danschef) [reply recipes/pandasgui/meta.yaml:25]: Ok, so I can safely remove the selectors in the complete recipe to make it noarch?

Comment (ocefpaf) [reply recipes/pandasgui/meta.yaml]: Let's try first without the CDTs, if we can fix this with only the yum requirements we can make Linux GUI happy and macOS and Windows can probably use the same noarch package. If we really need the CDTs we may need to drop the noarch.

Comment (danschef) [reply recipes/pandasgui/meta.yaml]: Ok, I did not convert it back to noarch, yet, but I removed the CDT packages you mentioned in https://github.com/conda-forge/staged-recipes/pull/13237#discussion_r524367625 and the libGL error does not show up again. So, they are obviously not needed.

However, there is still the `ImportError: libXcomposite.so.1`. According to [this](https://github.com/spyder-ide/spyder/issues/6990#issuecomment-418960594), a `sudo apt-get install python-pyqt5.qtwebkit` might help. But where do I have to add it?

Comment (danschef) [reply recipes/pandasgui/meta.yaml]: Adding python-pyqt5.qtwebkit to the host requirements to fix the `ImportError: libXcomposite.so.1` does not seem to work. Do I need to use a different package name?

**Comment by danschef** [comment]
@conda-forge/staged-recipes 
I think we need some assistance here. The Linux build still fails with `ImportError: libXcomposite.so.1` when importing QtWebEngineWidgets from PyQt5. So far, could not find a solution (see above).

File: recipes/pandasgui/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - pyqt 5
+    - python
+
+test:
+  imports:
+    - pandasgui
+    - pandasgui.widgets
+  commands:
+    - pip check
+  requires:
```
Comment (jakirkham) [inline recipes/pandasgui/meta.yaml:44]: Would add the same CDTs from `build` above here.

Comment (danschef) [reply recipes/pandasgui/meta.yaml:43]: @jakirkham @ocefpaf
Done, but without success.

Comment (adamerose) [reply recipes/pandasgui/meta.yaml:43]: Any advice on how to proceed?

Comment (danschef) [reply recipes/pandasgui/meta.yaml:43]: @conda-forge/staged-recipes

Comment (ocefpaf) [reply recipes/pandasgui/meta.yaml:43]: Sorry. Out of ideas. I can try a few things later this week but I cannot promise I'll have time for it.

Comment (danschef) [reply recipes/pandasgui/meta.yaml:43]: That would be very helpful, thanks!

**Comment by ocefpaf** [comment]
@adamerose you'll probably need https://github.com/conda-forge/pyqt-feedstock/blob/master/recipe/xcrun and https://github.com/conda-forge/pyqt-feedstock/blob/master/recipe/xcodebuild on macOS.

**Comment by ocefpaf** [comment]
Also, if you want to skip Windows and macOS for now, get this merged for Linux, we can fix the rest on the feedstock.

**Comment by danschef** [comment]
@ocefpaf Thanks a lot for looking into this! I would also suggest to merge this now and fix the rest on the feedstock. The recipe build seems to work fine on Linux and Windows, so only OSX is missing.
@adamrose If you like, I would offer to be a maintainer of the feedstock. Maybe I can support you there a bit.

**Comment by ocefpaf** [comment]
> @ocefpaf Thanks a lot for looking into this! I would also suggest to merge this now and fix the rest on the feedstock. The recipe build seems to work fine on Linux and Windows, so only OSX is missing.
> @adamrose If you like, I would offer to be a maintainer of the feedstock. Maybe I can support you there a bit.

Great, let's address the remaining issues on the feedstock. Apologies for the delay to get to this one but lately I've been fighting against my GH notifications a lot :smile:

**Comment by adamerose** [comment]
Thanks everyone! @danschef I added you to the maintainers list, would be great to have help since I'm new to using conda-forge.

**Pull request merged**